src/client/auth.js

```diff
@@ -1,16 +1,68 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('fs').promises
 
-module.exports = function (client, options) {
+module.exports = async function (client, options) {
   const yggdrasilClient = yggdrasil({ agent: options.agent, host: options.authServer || 'https://authserver.mojang.com' })
-  const clientToken = options.clientToken || (options.session && options.session.clientToken) || UUID.v4().toString()
+  const clientToken = options.clientToken || (options.session && options.session.clientToken) || (options.profilesFolder && (await getLauncherProfiles()).clientToken) || UUID.v4().toString().replace(/-/g, '')
   const skipValidation = false || options.skipValidation
   options.accessToken = null
-  options.haveCredentials = options.password != null || (clientToken != null && options.session != null)
+  options.haveCredentials = options.password != null || (clientToken != null && options.session != null) || options.profilesFolder != null
+
+  async function getLauncherProfiles () { // get launcher profiles
+    try {
+      return JSON.parse(await fs.readFile(options.profilesFolder + '/launcher_profiles.json', 'utf8'))
+    } catch (err) {
+      await fs.mkdir(options.profilesFolder, { recursive: true })
+      await fs.writeFile(options.profilesFolder + '/launcher_profiles.json', '{}')
```

Comment:
Will this mess with the launcher at all?

Reply:
Yes, 'mess' as in it will modify that file. It's compatible with the official launcher, which means it properly deletes _and_ also saves new accounts to the file for the launcher to use

Reply:
Also, one could just specify a path that is not the launcher folder and it will just use its own isolated file instead